sys/kern/sys_socket.c

```diff
@@ -598,7 +598,7 @@ soaio_process_job(struct socket *so, struct sockbuf *sb, struct kaiocb *job)
 
 	done = job->aio_done;
 	cnt = job->uaiocb.aio_nbytes - done;
-	IOVEC_INIT(&iov, (void *)((uintptr_t)job->uaiocb.aio_buf + done), cnt);
+	IOVEC_INIT(&iov, (void *)((__cheri_addr vaddr_t)job->uaiocb.aio_buf + done), cnt);
```

Comment:
I wonder what I was thinking when I wrote this code.  It should almost certainly be:
```C
IOVEC_INIT_C(&iov, (char * __capability)job->uaiocb.aio_buf + done, cnt);
```
It certainly should not be a provenance stripping cast.